src/lib/helpers.ts

```diff
@@ -8,3 +21,47 @@ export function getUsdcAddress(networkId: number): string {
 
   return buyTokenAddress;
 }
+
+export async function sendGaslessTransaction(contractCall: PromiEvent<Contract>): Promise<any> {
+  let hashOutcome = OUTCOMES.INITIAL;
+
+  const transactionHash = await new Promise(
+    (resolve, reject) => {
+      contractCall.on('error', (error: Error) => {
+        if (hashOutcome === OUTCOMES.INITIAL) {
+          hashOutcome = OUTCOMES.REJECTED;
+          reject(error);
+          (contractCall as any).off();
```

Comment:
What does this line do?

Reply:
I think it stops the contract call from waiting but Im not totally sure